src/telliot_feeds/reporters/reporter_autopay_utils.py

```diff
@@ -308,8 +308,13 @@ async def get_feed_tip(query_id: bytes, autopay: TellorFlexAutopayContract) -> A
     try:
         single_query = {query_id: CATALOG_QUERY_IDS[query_id]}
     except KeyError:
-        CATALOG_QUERY_IDS[query_id] = query_id.hex()
-        single_query = {query_id: CATALOG_QUERY_IDS[query_id]}
+        if "TellorRNG" in str(query_id):
```

Comment:
A couple things with this function:
- Looks like the function's return type could be changed from `Any` to `Optional[int]` or `Optional[float]`
- I don't understand the conditional on line 311. Why's it check if a query type ("TellorRNG") is in the stringified queryId? queryId would never contain the query type's name, as it's always going to be some has string like `abcde12341234aabbccddee123443...`. So this would always raise the exception on like 316. Unless `query_id` is not actually the queryId?
- Don't raise an exception that would break telliot on line 316. Instead, log an error & return None

Reply:
for the typing and the exception, I'll make that change.
for line 311, I also made a change on the rng reporter (https://github.com/tellor-io/telliot-feeds/blob/ce305644bb3b9806a1af9e94ffa32223f6dd03ec/src/telliot_feeds/reporters/rng_interval.py#L79-L81) so that the input is the querydata instead of the id, that way I can stringify it and check the query type. Since the rng generates a new query id each time and isn't going to be stored in the catalog and the submission through telliot is supported then the tip listener should check if there is a tip for it.

Reply:
Gotcha, I think instead of checking for specific query type names, a dynamic solution would be better, like:
```python
from clamfig.base import Registry
qtype_name, _ = decode(query_data)
if qtype_name not in Registry.registry:
    logger.warning(f"Unsupported query type: {qtype_name})
    return None
```